### ISSUE TITLE
View source of message

### ISSUE THREAD

**Issue by replaid** [Foundry376/Mailspring#441]
I expected to be able to view the source of my email messages.

It seems that the per-message dropdown menu would be a good place to put a View Source command.

##### Are there any related issues?

I could not find any issues or documentation mentioning this.

##### What operating system are you using?

PureOS 3

##### What version of Mailspring are you using?
1.0.10

##### Does this feature exist in another mail client or tool you use?

Yes, Apple Mail, Gmail on the web, MailMate, and every single other email client I've ever used on the desktop.

**Comment by bengotow** [comment]
You can technically do this by clicking in the blank space to the left of the dropdown:

![image](https://user-images.githubusercontent.com/1037212/34132992-b335f864-e40f-11e7-9edf-eb0043de84d3.png)

We'll definitely make this easier to find soon, it's just blocked on a bit of cleanup and making the items in the message list selectable. (So it could go in the application menu.)

**Comment by TakeMeNL** [comment]
_Small bump_
I use the 1.2.1 version and i don't have this button, i would love to see this feature because i use it alot for the spam emails in gmail 🎉

**Comment by sdpper** [comment]
It took me a bit of time to figure out where the "blank space" was located.  Found it between "reply-all" and the date/time of the message.  Talk about "hidden".  You guys did a good job hiding it for some unknown reason?  Not totally convinced MailSpring is a viable email client but will keep an open mind.  No pun intended....

**Comment by TakeMeNL** [comment]
@sdpper thx for pointing out where that button is. Indeed well hidden.

**Comment by sdpper** [comment]
@TakeMeNL your welcome!  I just uninstalled as MailSpring is not really ready for me.  They do have a wonderful head start however and if worked would have gladly paid.  Another issue is I could not find anywhere the ability to set a delete retention period on mail folders which is important.  Manually cleaning would be difficult for me.  So will stay with Thunderbird and maybe check them in a year or two.  Also their limit on 4 accounts was another show stopper.  Those kind of games are no fun.  They should provide a fully functional 30 day versus doing a limit as I'm not sure it could handle more than 4 accounts anyway.  Seem to take forever to sync.

**Comment by KaKi87** [comment]
Hello, any update on that feature ? Thanks

**Comment by dinoboy197** [comment]
I'm using Ubuntu 18.04, with Mailspring 1.4.2, and right-clicking in the white space between the reply button and date/time in an individual message does not show any kind of popup for looking at the message source.

**Comment by icfr** [comment]
@dinoboy197 it's a left-click

**Comment by dinoboy197** [comment]
Thank you @icfr ! That solves my issue. I do see the hidden menu now.

**Comment by phamhung77** [comment]
Just another request from me about having a button to save an email to source format, for example, a .eml file. I am using an external antispam filter, sometimes spam went through, and I need to train the system with new spam sample. 

In Mail app, it's straightforward to save any email to .eml format. It would be great if Mailspring has that feature too.

**Comment by CodeBrauer** [comment]
A menu item in "Thread" or a shortcut key would be nice too.

![Bildschirmfoto 2019-07-01 um 16 45 45](https://user-images.githubusercontent.com/2059754/60445539-b1a7e380-9c1f-11e9-9c17-5afd436925ee.png)

**Comment by samtuke** [comment]
This GitHub issue will rack up thousands of unnecessary views unless the 'view source' option is made visible somewhere. I don't see any reason for it to be concealed like this -- there is no risk to the user in viewing the source of their own emails.

**Comment by bluepangolin55** [comment]
I also don't understand why this 'essential' feature needs to be hidden in that way. It would be easy to solve it with a menu entry.

**Comment by CodeMouse92** [comment]
It appears that this issue has been resolved, so we are closing it. If the problem occurs again, please create a new thread in the appropriate category on our [Discourse](https://community.getmailspring.com).

We are in the process of migrating issues to Discourse, which can better facilitate discussion and discovery, and so GitHub Issues can focus on issues that are confirmed and slated for resolution in the near term. [Learn more about the changes here.](https://community.getmailspring.com/t/mailspring-community-transition/38)

@RealAct or @bluepangolin55 If you'd like to open a new Feature Suggestion on our Discourse in relation to Mailspring 1.8.0 or later (if the interface still isn't how you'd like it), we can revisit that topic separately: https://community.getmailspring.com

We hope to see you there soon!

-The Mailspring Team
